traffic_ops/traffic_ops_golang/riak.go

```diff
@@ -0,0 +1,267 @@
+
+		// create a storage object and store the data
+		obj := &riak.Object{
+			ContentType:     "text/json",
+			Charset:         "utf-8",
+			ContentEncoding: "utf-8",
+			Key:             xmlId,
+			Value:           []byte(data),
+		}
+
+		err = saveObject(obj, cdn_uri_keys_bucket, db, cfg)
+		if err != nil {
+			log.Errorf("%v\n", err)
+			handleErr(err, http.StatusInternalServerError)
+			return
+		}
+
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprintf(w, "%s", data)
+	}
+}
+
+// saves an object to riak storage
+func saveObject(obj *riak.Object, bucket string, db *sqlx.DB, cfg Config) error {
+	// create and start a cluster
+	cluster, err := getRiakCluster(db, cfg, 12)
+	if err != nil {
+		return err
+	}
+
+	defer func() {
+		if err := cluster.Stop(); err != nil {
+			log.Errorf("%v\n", err)
+		}
+	}()
+
+	if err = cluster.Start(); err != nil {
+		return err
+	}
+
+	// build store command and execute.
+	cmd, err := riak.NewStoreValueCommandBuilder().
+		WithBucket(bucket).
+		WithContent(obj).
+		Build()
+	if err != nil {
+		return err
+	}
+	if err := cluster.Execute(cmd); err != nil {
+		return err
+	}
+
+	return nil
+}
+
+// fetch an object from riak storage
+func fetchObject(key string, bucket string, db *sqlx.DB, cfg Config) (*riak.FetchValueCommand, error) {
+	// build the fetch command
+	cmd, err := riak.NewFetchValueCommandBuilder().
+		WithBucket(bucket).
+		WithKey(key).
+		Build()
+	if err != nil {
+		return nil, err
+	}
+	// create and start a riak cluster
+	cluster, err := getRiakCluster(db, cfg, 12)
+	if err != nil {
+		log.Errorf("%v\n", err)
+		return nil, err
+	}
+	defer func() {
+		if err := cluster.Stop(); err != nil {
+			log.Errorf("%v\n", err)
+		}
+	}()
+	if err = cluster.Start(); err != nil {
+		return nil, err
+	}
+	if err = cluster.Execute(cmd); err != nil {
+		return nil, err
+	}
+	fvc := cmd.(*riak.FetchValueCommand)
+
+	return fvc, err
+}
+
+// endpoint handler for fetching uri signing keys from riak
+func urisignkeysHandler(db *sqlx.DB, cfg Config) http.HandlerFunc {
+	return func(w http.ResponseWriter, r *http.Request) {
+		handleErr := func(err error, status int) {
+			log.Errorf("%v %v\n", r.RemoteAddr, err)
+			w.WriteHeader(status)
+			fmt.Fprintf(w, http.StatusText(status))
+		}
+
+		ctx := r.Context()
+		pathParams, err := getPathParams(ctx)
+		if err != nil {
+			handleErr(err, http.StatusInternalServerError)
+			return
+		}
+
+		xmlId := pathParams["xml-id"]
+
+		fvc, err := fetchObject(xmlId, cdn_uri_keys_bucket, db, cfg)
+		if err != nil {
+			handleErr(err, http.StatusInternalServerError)
+			return
+		}
+
+		resp, err := getStringValue(fvc.Response)
+		if err != nil {
+			handleErr(err, http.StatusNotFound)
+			return
+		}
+		w.Header().Set("Content-Type", "application/json")
+		fmt.Fprintf(w, "%s", resp)
+	}
+}
+
+// returns a riak cluster of online riak nodes.
+func getRiakCluster(db *sqlx.DB, cfg Config, maxNodes int) (*riak.Cluster, error) {
+	riakServerQuery := fmt.Sprintf(`
```

Comment:
This doesn't need to be a Sprintf, and it encourages people to add variables in the future, which is perilous for SQL Injection Vulnerabilities. Better to make it just the string.

Reply:
Yeah, I forgot to remove that.  Before I was using a parameter to build the sql and i eliminated it.  I'll update this